Src/Support/GoogleApis.Tests/Apis/Requests/Parameters/ParameterUtilsTest.cs

```diff
@@ -0,0 +1,74 @@
+﻿/*
+Copyright 2016 Google Inc
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+using System.Collections.Generic;
+
+using NUnit.Framework;
+
+using Google.Apis.Util;
+using Google.Apis.Requests;
+using Google.Apis.Requests.Parameters;
+
+namespace Google.Apis.Tests.Apis.Requests
+{
+    [TestFixture]
+    public class ParameterUtilsTest
+    {
+        private class TestRequestUrl
+        {
+            [Google.Apis.Util.RequestParameterAttribute("first_query_param", Google.Apis.Util.RequestParameterType.Query)]
```

Comment:
My bad. I wasn't clear before and I was actually *wrong*.
Seeing your code now, we actually allow users to set only RequestParameterType without name when we use query or path which is wrong! My mistake, you can revert the last change to the Attribute, and bring back the names to SecondParam and ParamsCollection.

When you bring back the name to GoogleAuthorizationCodeRequestUrl as well, add a comment that the name itself isn't being used for USerDefinedQuerieis.

If I'm not clear and during writing the change feel free to contact me directly, it's the same name I'm using with @google.com.
I'm sorry again, it's totally my bad.